### DISCUSSION TITLE
Enforce .all.each called when eager loading dataset

### DISCUSSION THREAD

**Discussion by TSMMark** [jeremyevans/sequel#2141] [General]
Forgetting to chain `.all` before `.each` when eager loading is one of the most common issues I encounter when reviewing or diagnosing issues newer teammates' Sequel code.

I did some quick searches and haven't found any plugin or option to raise an exception if `.each` is called without `.all`, or a rubocop rule.

Is that something that could be supported technically? Are you open to it? Any tips on how to implement?

Thanks!

**Comment by janko** [comment]
You're probably looking for the [eager_each](https://sequel.jeremyevans.net/rdoc-plugins/classes/Sequel/Plugins/EagerEach.html) plugin. We've used it a previous company and it solved this problem for us 🙂

**Reply by TSMMark** [reply]
This is great! Wonder why it didn't come up in search. Thank you
